File: tests/fixtures/Sample.java
package com.example;

import java.util.List;
import java.util.Optional;

public class Sample extends BaseService implements Serializable {

    private String dbPath;
    public static final int MAX_SIZE = 100;

    public Optional<Object> getUser(int userId) {
        return Optional.empty();
    }

    public List<Object> listUsers(int limit) {
        return List.of();
    }

    protected boolean validate(String input) {
        return true;
    }

    private void loadCache() {
    }
}

enum Status {
    ACTIVE,
    INACTIVE,
    PENDING
}

File: tests/fixtures/sample.py
"""Sample Python module for src2gdlc testing."""
import os
from pathlib import Path
from typing import Optional, List


class UserService:
    """Manages user operations."""

    def __init__(self, db_path: str):
        self.db_path = db_path

    def get_user(self, user_id: int) -> Optional[dict]:
        """Retrieve a user by ID."""
        pass

    def list_users(self, limit: int = 10) -> List[dict]:
        """List users with pagination."""
        pass

    def _validate(self, data: dict) -> bool:
        """Internal validation."""
        pass


def create_service(path: str) -> UserService:
    """Factory function."""
    return UserService(path)


STATUS_ACTIVE = "active"
STATUS_INACTIVE = "inactive"
